src/flash-source-buffer.js

```diff
@@ -169,6 +169,12 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
     };
 
     chunkInData();
+
+    // fire loaded metadata on first segment append
+    if (!this.firstAppendHappened) {
```

Comment:
Firing this after we have appended **all** of the data makes the most sense. It is better to fire these kind of events later rather than too early.